packages/core/src/middleware/koa-ui-guard.ts

```diff
@@ -0,0 +1,46 @@
+import { adminConsoleApplicationId } from '@logto/schemas/lib/seeds';
+import { MiddlewareType } from 'koa';
+import { IRouterParamContext } from 'koa-router';
+import { Provider } from 'oidc-provider';
+
+import { MountedApps } from '@/env-set';
+import { hasAdminUsers } from '@/queries/user';
+
+// Need To Align With UI
+export const sessionNotFoundPath = '/unknown-session';
+export const guardedPath = ['/sign-in', '/register', '/social/register'];
+
+export default function koaUiGuard<StateT, ContextT extends IRouterParamContext, ResponseBodyT>(
```

Comment:
maybe we've got some gap here. i would like to keep ui "pure" and guard (check admin users) in front of admin console routes. shall we discuss offline?

Reply:
yes let's line up early next week.  cc: @logto-io/eng  As my understanding is we have to block an actual  /sign-in request from ac sign-in flow, and force redirect, which somehow has to be a UI route scope.